subsys/net/lib/dns/dns_sd.c

```diff
@@ -789,7 +789,7 @@ int dns_sd_handle_ptr_query(const struct dns_sd_rec *inst, const struct in_addr
 	rsp->arcount++;
 	offset += r;
 
-	if (addr6 != NULL) {
+	if (addr6 != NULL && !net_ipv6_is_addr_unspecified(addr6)) {
```

Comment:
Are you sure that this is necessary? Is this just additional hardening? AFAIK, this code is only reachable after 
a) the local interface is up, and
b) after a query is received

For announcements on init / wake, (a) would need to be checked in advance - and the address that an announcement would need to be sent to would be the broadcast address (which is implicitly specified).

If anything, I would suggest putting `net_ipv6_is_addr_unspecified(addr6)` inside of the condition within an `__ASSERT()`. Same for ipv4. That would at least ensure that redundant code is compiled-out when assertions are disabled.

Also, please double-check that you have verified that the interface is up prior to calling `dns_sd_handle_ptr_query()` - IIRC, that was always the case before.